### ISSUE TITLE
Python module issues

### ISSUE THREAD

**Issue by matveitolochko** [DranikiRobotics/arc#3]
Submodules are currently broken.
See:
- https://github.com/PyO3/pyo3/issues/759
- https://github.com/PyO3/pyo3/issues/1517#issuecomment-808664021
- https://gist.github.com/ethanhs/fd4123487974c91c7e5960acc9aa2a77

**Comment by matveitolochko** [comment]
Fixed in bbc66fe3fd125faf548977f0ff92d1564688872f

Explanation:

1. We create the submodule via `PyModule::new`
2. Call that the function that represents the submodule (providing `py`, so this module)
3. Then we add the submodule to the parent module. (`m.add_submodule(module_name)`)
4. Finally, run a python script to add the submodule to `sys.modules`.

I created this macro to do so easier:
```rust
macro_rules! pymod {
    ($module_name: ident -> $submodule_func: path, $module_path: literal, $py:ident, $m: ident) => {
        let $module_name = PyModule::new($py, ::core::stringify!($module_name))?;
        $submodule_func($py, $module_name)?;
        $m.add_submodule($module_name)?;
        pyo3::py_run!(
            $py,
            $module_name,
            ::core::concat!(
                "import sys; sys.modules['",
                $module_path,
                "'] = ",
                ::core::stringify!($module_name),
            )
        );
    };
}
```

**Comment by matveitolochko** [comment]
Also added these same docs in 1774431def8ac13c25ffad0fbbf204ee2bece684
